src/ElfUtils/ElfFile.cpp

```diff
@@ -65,8 +67,47 @@ class ElfFileImpl : public ElfFile {
   std::string build_id_;
   bool has_symtab_section_;
   bool has_debug_info_section_;
+  std::optional<GnuDebugLinkInfo> gnu_debuglink_info_;
 };
 
+template <typename ElfT>
+ErrorMessageOr<GnuDebugLinkInfo> ReadGnuDebuglinkSection(
+    const typename ElfT::Shdr& section_header, const llvm::object::ELFFile<ElfT>& elf_file) {
+  llvm::Expected<llvm::ArrayRef<uint8_t>> contents_or_error =
+      elf_file.getSectionContents(&section_header);
+  if (!contents_or_error) return ErrorMessage("Could not obtain contents.");
+
+  const llvm::ArrayRef<uint8_t>& contents = contents_or_error.get();
+
+  if (contents.size() < 4) return ErrorMessage{"Section is too short."};
```

Comment:
is this supposed to be sizeof(ChecksumType)?

Reply:
and probably <= because of sizeof(path) is minimum 1 ('\0')

Reply:
Yes. Done. Thanks.